pinecone_text/dense/cohere_encoder.py

```diff
@@ -0,0 +1,107 @@
+from enum import Enum, StrEnum
+from typing import Union, List, Any, Optional
+from pinecone_text.dense.base_dense_ecoder import BaseDenseEncoder
+
+try:
+    import cohere
+    from cohere.error import CohereError
+except (OSError, ImportError, ModuleNotFoundError) as e:
+    _cohere_installed = False
+else:
+    _cohere_installed = True
+
+
+class CohereEncoderName(StrEnum):
+    """
+    Supported Cohere encoder models.
+    """
+    ENGLISH_V3 = "embed-english-v3.0"
+    ENGLISH_LIGHT_V3 = "embed-english-light-3.0"
+    MULTILINGUAL_V3 = "embed-multilingual-v3.0"
+    MULTILINGUAL_LIGHT_V3 = "embed-multilingual-light-v3.0"
+
+    @classmethod
+    def list_models(cls):
+        """Method to get a list of all model names."""
+        return [model.value for model in cls]
+
+class CohereInputType(Enum):
+    SEARCH_DOCUMENT = "search_document"
+    SEARCH_QUERY = "search_query"
+    # CLASSIFICATION = "classification"
+    # CLUSTERING = "clustering"
+
+
+class CohereEncoder(BaseDenseEncoder):
+    """
+    Cohere's text embedding wrapper. See https://docs.cohere.com/reference/embed
+
+    Note: You should provide an API key as the environment variable COHERE_API_KEY.
+          Or you can pass it as argument to the constructor as `api_key`.
+    """
+
+    def __init__(
+        self,
+        model_name: CohereEncoderName = CohereEncoderName.ENGLISH_V3,
+        api_key: Optional[str] = None,
+        **kwargs: Any,
+    ):
+        """
+        Initialize the Cohere encoder.
+
+        :param model_name: The name of the embedding model to use. See https://docs.cohere.com/reference/embed
+        :param kwargs: Additional arguments to pass to the underlying cohere client. See https://github.com/openai/openai-python
+        """
+        if not _cohere_installed:
+            raise ImportError(
+                "Failed to import cohere. Make sure you install cohere extra "
+                "dependencies by running: "
+                "`pip install pinecone-text[cohere]"
+            )
+        if model_name not in CohereEncoderName.list_models():
```

Comment:
I'm fine with letting Cohere API to handle this assertion, but we can also keep this assertion here

Reply:
yes I wanted to add in case someone tries to use `v2` model that doesn't include `input_type` params